Build failure: fbv

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

Build `fbv`.

### Can Hydra reproduce this build failure?

No, Hydra cannot reproduce this build failure.

### Link to Hydra build job

_No response_

### Relevant log output

```console
fbv> Running phase: unpackPhase
fbv> unpacking source archive /nix/store/b3pi8ygkpj4xzkd5s45gfwsj1043hjir-fbv-1.0b.tar.gz
fbv> source root is fbv-1.0b
fbv> setting SOURCE_DATE_EPOCH to timestamp 1094566655 of file "fbv-1.0b/Make.conf"
fbv> Running phase: patchPhase
fbv> applying patch /nix/store/p8j652vi91r1blsk95yjr5kc09zmwrnb-giflib-5.1.patch
fbv> patching file gif.c
fbv> Running phase: updateAutotoolsGnuConfigScriptsPhase
fbv> Running phase: configurePhase
fbv> patching script interpreter paths in ./configure
fbv> ./configure: interpreter directive changed from "#!/bin/sh" to "/nix/store/xy4jjgw87sbgwylm5kn047d9gkbhsr9x-bash-5.2p37/bin/sh"
fbv> configure flags: --prefix=/nix/store/9k56kkqcinwqi6casymx12d8g4z8jz64-fbv-1.0b
fbv> checking for libungif presence... no
fbv> checking for libjpeg presence... no
fbv> checking for libpng presence... no
fbv> building with bmp support... yes
fbv> checking for DEFAULT_FRAMEBUFFER... /dev/fb0
fbv> installation dir: /nix/store/9k56kkqcinwqi6casymx12d8g4z8jz64-fbv-1.0b/bin
fbv> manuals dir: /nix/store/9k56kkqcinwqi6casymx12d8g4z8jz64-fbv-1.0b/man
fbv> Running phase: buildPhase
fbv> build flags: -j16 SHELL=/nix/store/xy4jjgw87sbgwylm5kn047d9gkbhsr9x-bash-5.2p37/bin/bash LDFLAGS=-lgif
fbv> gcc  -O2 -Wall -D_GNU_SOURCE   -c -o main.o main.c
fbv> gcc  -O2 -Wall -D_GNU_SOURCE   -c -o jpeg.o jpeg.c
fbv> gcc  -O2 -Wall -D_GNU_SOURCE   -c -o gif.o gif.c
fbv> gcc  -O2 -Wall -D_GNU_SOURCE   -c -o png.o png.c
fbv> gcc  -O2 -Wall -D_GNU_SOURCE   -c -o bmp.o bmp.c
fbv> gcc  -O2 -Wall -D_GNU_SOURCE   -c -o fb_display.o fb_display.c
fbv> gcc  -O2 -Wall -D_GNU_SOURCE   -c -o transforms.o transforms.c
fbv> In file included from transforms.c:21:
fbv> transforms.c: In function 'simple_resize':
fbv> transforms.c:27:12: warning: suggest parentheses around assignment used as truth value []
fbv>    27 |     assert(cr = (unsigned char*) malloc(dx*dy*3));
fbv>       |            ^~
fbv> transforms.c: In function 'color_average_resize':
fbv> transforms.c:65:12: warning: suggest parentheses around assignment used as truth value []
fbv>    65 |     assert(cr=(unsigned char*) malloc(dx*dy*3)); p=cr;
fbv>       |            ^~
fbv> transforms.c: In function 'rotate':
fbv> transforms.c:93:16: warning: suggest parentheses around assignment used as truth value []
fbv>    93 |         assert(n = (unsigned char*) malloc(ox * oy * 3));
fbv>       |                ^
fbv> transforms.c: In function 'alpha_rotate':
fbv> transforms.c:143:16: warning: suggest parentheses around assignment used as truth value []
fbv>   143 |         assert(n = (unsigned char*) malloc(ox * oy));
fbv>       |                ^
fbv> bmp.c: In function 'fh_bmp_id':
fbv> bmp.c:51:9: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' []
fbv>    51 |         read(fd, id, 2);
fbv>       |         ^~~~~~~~~~~~~~~
fbv> bmp.c: In function 'fetch_pallete':
fbv> bmp.c:66:17: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' []
fbv>    66 |                 read(fd, buff, 4);
fbv>       |                 ^~~~~~~~~~~~~~~~~
fbv> bmp.c: In function 'fh_bmp_load':
fbv> bmp.c:89:9: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' []
fbv>    89 |         read(fd, buff, 4);
fbv>       |         ^~~~~~~~~~~~~~~~~
fbv> bmp.c:95:9: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' []
fbv>    95 |         read(fd, buff, 2);
fbv>       |         ^~~~~~~~~~~~~~~~~
fbv> bmp.c:104:41: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' []
fbv>   104 |                                         read(fd, buff, 1);
fbv>       |                                         ^~~~~~~~~~~~~~~~~
fbv> bmp.c:120:41: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' []
fbv>   120 |                                         read(fd, buff, 1);
fbv>       |                                         ^~~~~~~~~~~~~~~~~
fbv> bmp.c:136:41: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' []
fbv>   136 |                                         read(fd, buff, skip);
fbv>       |                                         ^~~~~~~~~~~~~~~~~~~~
fbv> bmp.c:147:41: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' []
fbv>   147 |                                         read(fd, buff, 1);
fbv>       |                                         ^~~~~~~~~~~~~~~~~
fbv> bmp.c:158:41: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' []
fbv>   158 |                                         read(fd, buff, 1);
fbv>       |                                         ^~~~~~~~~~~~~~~~~
fbv> bmp.c:165:41: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' []
fbv>   165 |                                         read(fd, buff, skip);
fbv>       |                                         ^~~~~~~~~~~~~~~~~~~~
fbv> bmp.c:176:41: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' []
fbv>   176 |                                         read(fd, buff, 1);
fbv>       |                                         ^~~~~~~~~~~~~~~~~
fbv> bmp.c:182:41: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' []
fbv>   182 |                                         read(fd, buff, skip);
fbv>       |                                         ^~~~~~~~~~~~~~~~~~~~
fbv> bmp.c:195:41: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' []
fbv>   195 |                                         read(fd, buff, 3);
fbv>       |                                         ^~~~~~~~~~~~~~~~~
fbv> bmp.c:201:41: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' []
fbv>   201 |                                         read(fd, buff, skip);
fbv>       |                                         ^~~~~~~~~~~~~~~~~~~~
fbv> bmp.c: In function 'fh_bmp_getsize':
fbv> bmp.c:226:9: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' []
fbv>   226 |         read(fd, size, 4);
fbv>       |         ^~~~~~~~~~~~~~~~~
fbv> bmp.c:229:9: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' []
fbv>   229 |         read(fd, size, 4);
fbv>       |         ^~~~~~~~~~~~~~~~~
fbv> jpeg.c: In function 'fh_jpeg_load':
fbv> jpeg.c:65:12: warning: variable 'py' set but not used []
fbv>    65 |     int px,py,c;
fbv>       |            ^~
fbv> jpeg.c: In function 'fh_jpeg_getsize':
fbv> jpeg.c:114:15: warning: variable 'c' set but not used []
fbv>   114 |     int px,py,c;
fbv>       |               ^
fbv> jpeg.c: In function 'fh_jpeg_id':
fbv> jpeg.c:43:5: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' []
fbv>    43 |     read(fd,id,10);
fbv>       |     ^~~~~~~~~~~~~~
fbv> gif.c: In function 'fh_gif_id':
fbv> gif.c:45:5: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' []
fbv>    45 |     read(fd,id,4);
fbv>       |     ^~~~~~~~~~~~~
fbv> png.c: In function 'fh_png_load':
fbv> png.c:108:31: warning: pointer targets in assignment from 'unsigned char *' to 'char *' differ in signedness []
fbv>   108 |                         fbptr = buffer;
fbv>       |                               ^
fbv> png.c:114:54: warning: pointer targets in initialization of 'unsigned char *' from 'char *' differ in signedness []
fbv>   114 |                                 unsigned char *trp = rp;
fbv>       |                                                      ^~
fbv> png.c:131:31: warning: pointer targets in assignment from 'unsigned char *' to 'char *' differ in signedness []
fbv>   131 |                         fbptr = buffer;
fbv>       |                               ^
fbv> png.c: In function 'fh_png_id':
fbv> png.c:41:5: warning: ignoring return value of 'read' declared with attribute 'warn_unused_result' []
fbv>    41 |     read(fd,id,4);
fbv>       |     ^~~~~~~~~~~~~
fbv> main.c: In function 'show_image':
fbv> main.c:433:17: warning: 'i.rgb' may be used uninitialized []
fbv>   433 |                 free(i.rgb);
fbv>       |                 ^~~~~~~~~~~
fbv> main.c:210:22: note: 'i.rgb' was declared here
fbv>   210 |         struct image i;
fbv>       |                      ^
fbv> main.c:434:17: warning: 'i.alpha' may be used uninitialized []
fbv>   434 |                 free(i.alpha);
fbv>       |                 ^~~~~~~~~~~~~
fbv> main.c:210:22: note: 'i.alpha' was declared here
fbv>   210 |         struct image i;
fbv>       |                      ^
fbv> main.c:431:11: warning: 'i.do_free' may be used uninitialized []
fbv>   431 |         if(i.do_free)
fbv>       |           ^
fbv> main.c:210:22: note: 'i.do_free' was declared here
fbv>   210 |         struct image i;
fbv>       |                      ^
fbv> gcc  -lgif -o fbv main.o jpeg.o gif.o png.o bmp.o fb_display.o transforms.o
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: jpeg.o: in function `fh_jpeg_load':
fbv> jpeg.c:(.text+0x110): undefined reference to `jpeg_std_error'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: jpeg.c:(.text+0x154): undefined reference to `jpeg_CreateDecompress'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: jpeg.c:(.text+0x160): undefined reference to `jpeg_stdio_src'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: jpeg.c:(.text+0x16d): undefined reference to `jpeg_read_header'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: jpeg.c:(.text+0x180): undefined reference to `jpeg_start_decompress'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: jpeg.c:(.text+0x1e1): undefined reference to `jpeg_read_scanlines'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: jpeg.c:(.text+0x214): undefined reference to `jpeg_finish_decompress'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: jpeg.c:(.text+0x21c): undefined reference to `jpeg_destroy_decompress'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: jpeg.c:(.text+0x259): undefined reference to `jpeg_destroy_decompress'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: jpeg.o: in function `fh_jpeg_getsize':
fbv> jpeg.c:(.text+0x2c5): undefined reference to `jpeg_std_error'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: jpeg.c:(.text+0x309): undefined reference to `jpeg_CreateDecompress'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: jpeg.c:(.text+0x319): undefined reference to `jpeg_stdio_src'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: jpeg.c:(.text+0x326): undefined reference to `jpeg_read_header'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: jpeg.c:(.text+0x339): undefined reference to `jpeg_start_decompress'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: jpeg.c:(.text+0x35d): undefined reference to `jpeg_destroy_decompress'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: jpeg.c:(.text+0x3a1): undefined reference to `jpeg_destroy_decompress'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.o: in function `fh_png_load':
fbv> png.c:(.text+0xd4): undefined reference to `png_create_read_struct'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0xea): undefined reference to `png_create_info_struct'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x125): undefined reference to `png_init_io'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x134): undefined reference to `png_read_info'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x165): undefined reference to `png_get_IHDR'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x1a6): undefined reference to `png_get_valid'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x1c5): undefined reference to `png_set_strip_16'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x1cf): undefined reference to `png_set_interlace_handling'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x1e2): undefined reference to `png_read_update_info'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x248): undefined reference to `png_read_rows'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x278): undefined reference to `png_read_end'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x289): undefined reference to `png_destroy_read_struct'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x2a6): undefined reference to `png_destroy_read_struct'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x393): undefined reference to `png_read_rows'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x418): undefined reference to `png_set_gray_to_rgb'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x438): undefined reference to `png_destroy_read_struct'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x451): undefined reference to `png_set_packing'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x45b): undefined reference to `png_set_tRNS_to_alpha'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x46d): undefined reference to `png_set_expand'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.o: in function `fh_png_getsize':
fbv> png.c:(.text+0x4d7): undefined reference to `png_create_read_struct'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x4e9): undefined reference to `png_create_info_struct'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x518): undefined reference to `png_destroy_read_struct'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x56e): undefined reference to `png_init_io'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x57d): undefined reference to `png_read_info'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x5ae): undefined reference to `png_get_IHDR'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x5c3): undefined reference to `png_destroy_read_struct'
fbv> /nix/store/v63bxfiacw082c7ijshf60alvvrpfxsq-binutils-2.44/bin/ld: png.c:(.text+0x5fa): undefined reference to `png_destroy_read_struct'
fbv> collect2: error: ld returned 1 exit status
fbv> make: *** [Makefile:21: fbv] Error 1
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.90, NixOS, 24.11 (Vicuna), 24.11.717984.ba8b70ee098b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.14`
 - channels(root): `"nixos-24.11, nixos-hardware, nixos-unstable"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### Notify maintainers

@peterhoeg

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/fb/fbv/package.nix
+++ b/pkgs/by-name/fb/fbv/package.nix
@@ -1,39 +1,28 @@
 {
   lib,
   stdenv,
-  fetchurl,
-  fetchpatch,
+  fetchFromGitHub,
   getopt,
   libjpeg,
   libpng12,
-  giflib,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "fbv";
-  version = "1.0b";
+  version = "1.0c";
 
-  src = fetchurl {
-    url = "http://s-tech.elsat.net.pl/fbv/fbv-${version}.tar.gz";
-    sha256 = "0g5b550vk11l639y8p5sx1v1i6ihgqk0x1hd0ri1bc2yzpdbjmcv";
+  src = fetchFromGitHub {
+    owner = "jstkdng";
+    repo = "fbv";
+    tag = finalAttrs.version;
+    hash = "sha256-4tAIFklKsx2uI+FQjq9vdolYm6d6YWugioG6k2ZUMrs=";
   };
-
-  patches = [
-    (fetchpatch {
-      url = "https://raw.githubusercontent.com/void-linux/void-packages/4a5bfe522ea5afd8203e804dc6a642d0871cd6dd/srcpkgs/fbv/patches/giflib-5.1.patch";
-      sha256 = "00q1zcn92yvvyij68bnq0m1sr3a411w914f4nyp6mpz0j5xc6dc7";
-    })
-  ];
-
-  patchFlags = [ "-p0" ];
 
   buildInputs = [
     getopt
     libjpeg
     libpng12
-    giflib
   ];
-  makeFlags = [ "LDFLAGS=-lgif" ];
 
   enableParallelBuilding = true;
 
@@ -41,11 +30,11 @@ stdenv.mkDerivation rec {
     mkdir -p $out/{bin,man/man1}
   '';
 
-  meta = with lib; {
+  meta = {
     description = "View pictures on a linux framebuffer device";
-    homepage = "http://s-tech.elsat.net.pl/fbv/";
-    license = licenses.gpl2Only;
-    maintainers = with maintainers; [ peterhoeg ];
+    homepage = "https://github.com/jstkdng/fbv";
+    license = lib.licenses.gpl2Only;
+    maintainers = with lib.maintainers; [ peterhoeg ];
     mainProgram = "fbv";
   };
-}
+})

